ambari-server/src/main/resources/common-services/AMBARI_INFRA_SOLR/0.1.0/package/scripts/collection.py

```diff
@@ -0,0 +1,127 @@
+    - solr_collection: collection to backup (that will be used in the BACKUP rest api command)
+    - solr_shared_fs: if this flag is true, then it will generate specific folder for every backup with a hostname suffix
+    where "." chars replaced with "_"(e.g.: /my/path/backup_locationc7301_ambari_apache_org), that can be useful if different
+    hosts share the same filesystem where the backup is stored.
+    """
+    import params
+    env.set_params(params)
+
+    backup_collection = default("/commandParams/solr_collection", "ranger_audits")
+    backup_location = default("/commandParams/solr_backup_location", None)
+    backup_name = default("/commandParams/solr_backup_name", "ranger")
+    shared_fs = default("/commandParams/solr_shared_fs", False)
+
+    solr_protocol = "https" if params.infra_solr_ssl_enabled else "http"
+    solr_base_url = format("{solr_protocol}://{hostname}:{infra_solr_port}/solr")
+
+    keytab = params.infra_solr_kerberos_keytab
+    principal = params.infra_solr_kerberos_principal
+
+    hostname_suffix = params.hostname.replace(".", "_")
+
+    if shared_fs:
+      backup_location = format("{backup_location}_{hostname_suffix}")
+
+    Logger.info(format("Backup Solr Collection {backup_collection} to {backup_location}"))
+
+    solr_url = format("{solr_base_url}/{backup_collection}/replication?command=BACKUP&location={backup_location}&name={backup_name}&wt=json")
+    backup_api_cmd = format("kinit -kt {keytab} {principal} && curl -k --negotiate -u : '{solr_url}'") if params.security_enabled else format("curl -k '{solr_url}'")
+
+    Directory(backup_location,
+              mode=0755,
+              cd_access='a',
+              owner=params.infra_solr_user,
+              group=params.user_group
+              )
+
+    Execute(backup_api_cmd, user=params.infra_solr_user, logoutput=True)
+
+def restore_collection(env):
+    """
+    Restore collections using replication API (as Solr Cloud Backup API is not available in Solr 5)
+    - solr_backup_name: it will be used in the snapshot name, if it's ranger, the snapshot folder will be snapshot.ranger
+    - solr_backup_location: folder location which contains the snapshot folder
+    - solr_collection: collection to backup (that will be used in the BACKUP rest api command)
+    - solr_shared_fs: if this flag is true, then it will generate specific folder for every backup with a hostname suffix
+    where "." chars replaced with "_"(e.g.: /my/path/backup_locationc7301_ambari_apache_org), that can be useful if different
+    hosts share the same filesystem where the backup is stored.
+    """
+    import params
+    env.set_params(params)
+
+    backup_collection = default("/commandParams/solr_collection", "ranger_audits")
+    backup_location = default("/commandParams/solr_backup_location", None)
+    backup_name = default("/commandParams/solr_backup_name", "ranger")
+    shared_fs = default("/commandParams/solr_shared_fs", False)
```

Comment:
Many variables and their initialization are the same like in the backup_collection function. Is it possible to extract them into a shared .py like params.py?

Reply:
its importing params.py, i felt like its better if that code part can be there in the command methods for readability, especially because the docs for the commands are there at place. (also to not define this variables for start / stop, otherwise i will need to use a bit longer variable names to make things clear), most of them occurs in 2 methods, the migration script param names are a bit different, as its not the last version of the patch probably i can include something like that (with longer more specific variable names)

Reply:
It does not have to be params.py it can be for example collection_params.py.